python/cudf/cudf/dataframe/dataframe.py

```diff
@@ -1569,6 +1569,7 @@ def one_hot_encoding(
         3         4      bird          0          1.0          0.0          0.0
         4         5      fish          2          0.0          0.0          1.0
         """
+        cats = list(cats)
```

Comment:
Instead of a list we should be able to use a Pandas Series here. I also believe there's another place we're doing a similar `for cat in cats` loop that should be addressed.

Reply:
Doing it once here should suffice, no? Because we are propagating this same `cats` into `series.one_hot_encoding` where we do `for cat in cats`. Or you prefer converting to Pandas at two places?

Reply:
Made the change at both places as we can have the same perf increase while using series api too. Updated the perf stats after merging the other copy related perf changes from master 🎉